tests/drivers/ipm/src/main.c

```diff
@@ -32,9 +32,9 @@ extern struct ipm_driver_api ipm_dummy_api;
 
 /* Set up the dummy IPM driver */
 struct ipm_dummy_driver_data ipm_dummy0_driver_data;
-DEVICE_DEFINE(ipm_dummy0, "ipm_dummy0", NULL,
+DEVICE_DEFINE(ipm_dummy0, "ipm_dummy0", ipm_dummy_init,
```

Comment:
why is a dummy init call needed?

Reply:
to check the dummy device for this test that it pass init stage

Reply:
Device successful init needs to be tested by getting a valid reference when using device_get_binding, not by inserting dummy calls

Reply:
What is the problem to check the init this way for this test's purposes ? 
(device_get_binding() check is added as well)